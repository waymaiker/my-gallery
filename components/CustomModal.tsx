import { Modal, StyleSheet, View } from "react-native";

type Props = {
  children: any,
  isVisible: boolean,
  setVisibility: Function
}

export default function CustomModal({children, isVisible, setVisibility}: Props){
  return <Modal
    animationType="fade"
    transparent={true}
    visible={isVisible}
    onRequestClose={() => setVisibility(!isVisible)}
  >
    <View style={styles.container}>
      <View style={styles.modalView}>
        {children}
      </View>
    </View>
  </Modal>
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    alignItems: 'center',
    justifyContent: 'center',
  },
  modalView: {
    width: "80%",
    justifyContent: 'space-around',
    borderRadius: 10,
  },
});