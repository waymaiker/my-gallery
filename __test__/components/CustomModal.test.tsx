import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import CustomModal from '../../components/CustomModal';
import { View } from 'react-native';

const onClick = jest.fn();
const Children = () => (<View></View>);

describe('<CustomModal />', () => {
  it('should match the snapshot', () => {    
    const rendered = render(<CustomModal isVisible={true} setVisibility={onClick} children={<Children />}/>);
    expect(rendered).toMatchSnapshot();
  });
})