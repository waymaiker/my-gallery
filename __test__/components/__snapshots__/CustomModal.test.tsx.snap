// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomModal /> should match the snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 10,
          "justifyContent": "space-around",
          "width": "80%",
        }
      }
    >
      <View />
    </View>
  </View>
</Modal>
`;
